File: config/config-sbc2ha-new/src/main/java/com/dfi/sbc2ha/config/sbc2ha/definition/sensor/ModbusSensorConfig.java
package com.dfi.sbc2ha.config.sbc2ha.definition.sensor;

import com.dfi.sbc2ha.config.sbc2ha.definition.enums.PlatformType;
import lombok.Data;
import lombok.EqualsAndHashCode;

@EqualsAndHashCode(callSuper = true)
@Data
public class ModbusSensorConfig extends BusSensorConfig {

    int address;
    String model;

    public ModbusSensorConfig() {
        super();
        platform = PlatformType.MODBUS;
    }
}

File: config/editor/src/app/definition/sensor/ModbusSensorConfig.ts
import {BusSensorConfig} from "./BusSensorConfig";
import {PlatformType} from "../enums/PlatformType";


export class ModbusSensorConfig extends BusSensorConfig {

    address: number;
    model: string;

    constructor(name: string, busId: string, address: number, model: string) {
        super(PlatformType.MODBUS, name, busId);
        this.address = address;
        this.model = model;
    }
}
